developerportal/apps/bakery/views.py

```diff
@@ -34,7 +34,7 @@ def build_object(self, obj):
         Build wagtail page and set SERVER_NAME to retrieve corresponding site
         object.
         """
-        site = obj.get_site()
+        site = obj.specific.get_site()
```

Comment:
`.specific` lets us call `get_site()` on our Page subclass so we can use our fallback solution